handlers/main.yml

```diff
@@ -96,13 +96,18 @@
            bond_master_interfaces_changed }}
     all_interfaces_changed: "{{ all_interfaces_changed_map[ansible_os_family] }}"
   notify:
+    - Pause playbook to give interfaces time to recover
     - Gather facts
     - Verify network interfaces exist
     - Verify network interfaces are active
     - Verify network interface IP configuration
     - Verify bond interface slaves
     - Verify bridge interface ports
 
+- name: Pause playbook to give interfaces time to recover
+  pause:
+    seconds: "{{ playbook_pause_time }}"
```

Comment:
To which race might you be referring to? The idea behind this was simply to avoid hitting the gather_facts tasks before the interfaces were up on the server. If there's a better way of doing this I'm totally open to it. What I was seeing was the interfaces weren't all up by the time facts were gathered, resulting in failures when checking for active interfaces.

Reply:
I guess my point was, if we need to sleep here, that suggests to me there is a race condition between the network being 'up' and us gathering facts. Adding a sleep isn't necessarily going to guarantee that we gather facts after the interfaces are up - how do we know how long to sleep for? Would that value work everywhere?

Perhaps the default of 3 seconds is sufficient to work in most environments. It's not something I've needed to do, but then I primarily use CentOS.